File: 1677-matrix-diagonal-sum/matrix-diagonal-sum.cpp
class Solution {
public:
    int diagonalSum(vector<vector<int>>& mat) {
        int n = mat.size();
        int Sum1 = 0;
        int Sum2 = 0;

        for (int i = 0; i < n; i++) {
            Sum1 += mat[i][i];
            Sum2 += mat[i][n - i - 1];
        }

        if (n % 2 != 0) {
            Sum1 -= mat[n / 2][n / 2];
        }
        
        return Sum1 + Sum2;
    }
};

File: 1677-matrix-diagonal-sum/matrix-diagonal-sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        diagonalSum, n = 0, len(mat)
        for i in range(n):
            diagonalSum += mat[i][i]
            diagonalSum += mat[i][n - 1 - i]
        if n % 2 == 1:
            diagonalSum -= mat[n // 2][n // 2]
        return diagonalSum